src/components/balances/token-asset-list-item.tsx

```diff
@@ -1,26 +1,44 @@
-import * as React from 'react';
-import { AddressBalanceResponse } from '@stacks/stacks-blockchain-api-types';
+import React, { useEffect, useState } from 'react';
+import BigNumber from 'bignumber.js';
+import { useAtomValue } from 'jotai/utils';
+import { FungibleTokenMetadata } from '@stacks/blockchain-api-client';
 import { Box, color, DynamicColorCircle, Flex, FlexProps, Stack } from '@stacks/ui';
 import { Caption, Text } from '@components/typography';
 import { TxLink } from '@components/links';
-import { getAssetNameParts } from '@common/utils';
+import { ftDecimals, getAssetNameParts } from '@common/utils';
 import { getTicker } from '@components/tx-events';
+import { apiClientsState } from '@store/api-clients';
 
 interface TokenAssetListItemProps extends FlexProps {
+  balanceOrCount: string;
```

Comment:
Reusing the same prop for different values is best avoided if possible. You could create a conditional, discriminating type here:

```ts
interface TokenAssetListItemFungibleProps extends FlexProps {
  tokenType: 'fungible_tokens';
  balance: string;
  token: string;
}
interface TokenAssetListItemNonFungibleProps extends FlexProps {
  tokenType: 'non_fungible_tokens';
  count: string;
  token: string;
}
type TokenAssetListItemProps = TokenAssetListItemFungibleProps | TokenAssetListItemNonFungibleProps;
```

Reply:
Yep, it was originally just `balance` and I renamed it. I'll look at doing your suggestion. I was just trying to keep this simple until someone is really working on the explorer consistently again and can refactor a lot of this code.

Reply:
You could maybe use the word 'amount' instead of balanceOrCount (that's what we do in the wallet) and make this view pure. The calculation logic can be done outside and then only display the values passed to this view component (like AssetItem in the wallet)

Reply:
Ok, thx for the suggestion, I'll check out the web wallet implementation.